src/goto-programs/goto_convert.cpp

```diff
@@ -650,12 +629,10 @@ void goto_convertt::convert_decl(
 {
   const exprt &op = code.symbol();
 
-  if(op.id() != ID_symbol)
-  {
-    error().source_location = op.find_source_location();
-    error() << "decl statement expects symbol as operand" << eom;
-    throw 0;
-  }
+  INVARIANT(
+    op.id() == ID_symbol,
+    op.find_source_location().as_string() +
+      ": decl statement expects symbol as operand");
```

Comment:
I think this ought to be moved to `std_code.h`, as part of the sanity checks of `to_code_declt`, and the return type of the `.symbol()` function should become `symbol_exprt`.